### PR TITLE
Ensure to load only regular files for CAs

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Ensure to load only regular files for CAs

## Motivation and Context
In kubernetes statefulset like environments when secrets
are mounted to pods they have sub-directories, we should
ideally be only looking for regular files here and skip
all others.

## How Has This Been Tested?
Manually using kubernetes statefulset deployment.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by xdire — APPROVED** [review]
Great thing to have

**Review by nitisht — DISMISSED** [review]
LGTM

**Comment by harshavardhana** [comment]
Don't merge this yet.. still working with the user.

**Comment by kannappanr** [comment]
marked as blocked as this PR is still undergoing changes. will unblock it once @harshavardhana says so

**Comment by harshavardhana** [comment]
Removing blocked as this is finally tested and validated correctly.

**Review by nitisht — APPROVED** [review]
LGTM and tested with self-signed certs

**Review by nitisht — CHANGES_REQUESTED** [review]
`CAs` directory is not mounted, so self signed certs not working at the moment. Investigating now.

**Review by nitisht — APPROVED** [review]
LGTM & tested

**Pull request merged**
